src/core/unittest/TicketTest.cpp

```diff
@@ -208,6 +540,457 @@ TEST(ResumptionTicketTest, ServerEncDecNoAppData)
+            &DecodedTP,
+            &DecodedAppData,
+            &DecodedAppDataLength));
+    InputTicketBuffer[0] = CXPLAT_TLS_RESUMPTION_CLIENT_TICKET_VERSION;
+
+    // Unsupported QUIC version
+    InputTicketBuffer[1] = 1;
+    InputTicketBuffer[2] = 1;
+    InputTicketBuffer[3] = 1;
+    InputTicketBuffer[4] = 1;
+    ASSERT_EQ(
+        QUIC_STATUS_INVALID_PARAMETER,
+        QuicCryptoDecodeServerTicket(
+            nullptr,
+            ActualEncodedTicketLength,
+            InputTicketBuffer,
+            AlpnList,
+            sizeof(AlpnList),
+            &DecodedTP,
+            &DecodedAppData,
+            &DecodedAppDataLength));
+    InputTicketBuffer[1] = 0;
+    InputTicketBuffer[2] = 0;
+    InputTicketBuffer[3] = 0;
+    InputTicketBuffer[4] = 1;
+
+    // Negotiated ALPN length shorter than actual
+    for (uint8_t s = 0; s < (uint8_t)sizeof(Alpn); ++s) {
+        QuicTraceLogInfo(
+            ServerResumptionTicketDecodeFailAlpnLengthShort,
+            "[test] Attempting to decode Negotiated ALPN with length %u (Actual: %u)",
+            s,
+            (uint8_t)sizeof(Alpn));
+
+        InputTicketBuffer[5] = s;
+        ASSERT_EQ(
+            QUIC_STATUS_INVALID_PARAMETER,
+            QuicCryptoDecodeServerTicket(
+                nullptr,
+                ActualEncodedTicketLength,
+                InputTicketBuffer,
+                AlpnList,
+                sizeof(AlpnList),
+                &DecodedTP,
+                &DecodedAppData,
+                &DecodedAppDataLength));
+    }
+
+    // Negotiated ALPN length longer than actual
+    InputTicketBuffer[5] = (uint8_t)sizeof(Alpn) + 1;
+    ASSERT_EQ(
+        QUIC_STATUS_INVALID_PARAMETER,
+        QuicCryptoDecodeServerTicket(
+            nullptr,
+            ActualEncodedTicketLength,
+            InputTicketBuffer,
+            AlpnList,
+            sizeof(AlpnList),
+            &DecodedTP,
+            &DecodedAppData,
+            &DecodedAppDataLength));
+
+    // Negotiated ALPN length improperly encoded QUIC_VAR_INT
+    for (uint8_t i = 1; i < 4; ++i) {
+        InputTicketBuffer[5] = (uint8_t)(i << 6);
+
+        QuicTraceLogInfo(
+            ServerResumptionTicketDecodeFailAlpnLengthEncodedWrong,
+            "[test] Attempting to decode Negotiated ALPN length (improperly encoded) %x (Actual: %u)",
+            InputTicketBuffer[5],
+            (uint8_t)sizeof(Alpn));
+
+        ASSERT_EQ(
+            QUIC_STATUS_INVALID_PARAMETER,
+            QuicCryptoDecodeServerTicket(
+                nullptr,
+                ActualEncodedTicketLength,
+                InputTicketBuffer,
+                AlpnList,
+                sizeof(AlpnList),
+                &DecodedTP,
+                &DecodedAppData,
+                &DecodedAppDataLength));
+    }
+    InputTicketBuffer[5] = (uint8_t)sizeof(Alpn);
+
+    // Handshake TP length shorter than actual
+    for (uint8_t s = 0; s < EncodedTPLength - CxPlatTlsTPHeaderSize; ++s) {
```

Comment:
```suggestion
    for (uint32_t s = 0; s < EncodedTPLength - CxPlatTlsTPHeaderSize; ++s) {
```
Fix this Code scanning failure please?